src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -117,6 +117,20 @@ public function normalize($object, $format = null, array $context = [])
                 $attributeValue = $callback($attributeValue, $object, $attribute, $format, $context);
             }
 
+            if (isset($attributesMetadata[$attribute]) && !empty($attributesMetadata[$attribute]->getEmbedProperties())) {
```

Comment:
No point for `empty` as it will be handled by `foreach` anyway.

Reply:
If you remove this empty you can move `continue 2` into loop.

Reply:
You are right, I will remove the `empty` :+1:

But the `continue` need to be out of the loop to treat all embedProperties, no ?

Reply:
:thinking: Without the `empty` check, I'll do the `continue`, that is not wanted :slightly_frowning_face:

Reply:
`getEmbedProperties()` is called two times on the hot path. As method calls are costly in PHP, you should store the return in a variable and reuse it.

Reply:
Fixed